block-storage/src/test/scala/coop/rchain/blockstorage/BlockStoreTest.scala

```diff
@@ -33,78 +43,96 @@ trait BlockStoreTest
   ): (BlockHash, BlockMessage) =
     (ByteString.copyFromUtf8(s._1), s._2)
 
-  def withStore[R](f: BlockStore[Id] => R): R
+  def withStore[R](f: BlockStore[Task] => Task[R]): R
 
   "Block Store" should "return Some(message) on get for a published key" in {
-    withStore { store =>
-      forAll(blockHashElementsGen, minSize(0), sizeRange(10)) { blockStoreElements =>
+    forAll(blockHashElementsGen, minSize(0), sizeRange(10)) { blockStoreElements =>
+      withStore { store =>
         val items = blockStoreElements
-        items.foreach(store.put(_))
-        items.foreach {
-          case (k, v) =>
-            store.get(k) shouldBe Some(v)
-        }
-        store.asMap().size shouldEqual items.size
-        store.clear()
+        for {
+          _ <- items.traverse_(store.put(_))
+          _ <- items.traverse[Task, Assertion] {
+                case (k, v) =>
+                  store.get(k).map(_ shouldBe Some(v))
+              }
+          result <- store.find(_ => true).map(_.size shouldEqual items.size)
+          _      <- store.clear()
```

Comment:
I don't think you need to clear the store if you're creating a new store for every generated `blockStoreElements`